src/app/shared/auth-guard.service.ts

```diff
@@ -43,15 +43,27 @@ export class AuthService {
     );
   }
 
+  private requiresNoAuth(route: ActivatedRouteSnapshot): boolean {

```

Comment:
This function flips the current logic which previously traveled up the route's parent along the tree to now going down the route's children. We should keep it to parents because allowing a child route to overwrite a parent's auth requirement is an easy way to end up with unintended side effects when opening up pages.